designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/LocationSearchBar.kt

```diff
@@ -0,0 +1,390 @@
+            }
+
+            item {
+                Row(
+                    modifier = Modifier.fillMaxWidth().padding(8.dp),
+                ) {
+                    Button(
+                        modifier = Modifier.fillMaxWidth(),
+                        style = ButtonStyle.TONAL,
+                        icon = {
+                            Icon(
+                                imageVector = Icons.Outlined.Map,
+                                contentDescription = "touch app",
+                            )
+                        },
+                        text = provideStringResource("select_in_map"),
+                        onClick = onBackClicked,
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun LocationItem(
+    modifier: Modifier = Modifier,
+    locationItemModel: LocationItemModel,
+    icon: @Composable () -> Unit,
+    onClick: () -> Unit,
+) {
+    Row(
+        modifier = modifier
+            .fillMaxWidth()
+            .clip(Shape.Small)
+            .clickableWithRipple(onClick = onClick)
+            .padding(Spacing.Spacing8),
+        horizontalArrangement = spacedBy(Spacing.Spacing16),
+        verticalAlignment = Top,
+    ) {
+        icon()
+
+        Column(modifier = Modifier.fillMaxWidth()) {
+            Text(
+                text = locationItemModel.title,
+                style = DHIS2SCustomTextStyles.titleMediumBold,
+                color = TextColor.OnPrimaryContainer,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis,
+            )
+            Text(
+                text = locationItemModel.subtitle,
+                style = MaterialTheme.typography.bodySmall,
+                color = TextColor.OnSurface,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis,
+            )
+        }
+    }
+}
+
+@Composable
+private fun SearchResultLocationItem(
+    modifier: Modifier = Modifier,
+    locationItemModel: LocationItemModel,
+    onClick: () -> Unit,
+) {
+    val icon = when (locationItemModel) {
+        is LocationItemModel.StoredResult -> Icons.Outlined.WatchLater
+        is LocationItemModel.SearchResult -> Icons.Outlined.Place
+    }
+    val tintedColor = when (locationItemModel) {
+        is LocationItemModel.StoredResult -> SurfaceColor.Warning
+        is LocationItemModel.SearchResult -> SurfaceColor.Primary
+    }
+
+    val bgColor = when (locationItemModel) {
+        is LocationItemModel.StoredResult -> SurfaceColor.WarningContainer
+        is LocationItemModel.SearchResult -> SurfaceColor.PrimaryContainer
+    }
+
+    LocationItem(
+        modifier = modifier,
+        locationItemModel = locationItemModel,
+        icon = {
+            LocationItemIcon(
+                icon = icon,
+                tintedColor = tintedColor,
+                bgColor = bgColor,
+            )
+        },
+        onClick = onClick,
+    )
+}
+
+@Composable
+fun LocationItemIcon(
```

Comment:
Could this function be private?

Reply:
I've decided to leave it public so that other developers can add new/custom location icons